crates/autopilot/src/domain/eth/mod.rs

```diff
@@ -205,3 +205,20 @@ pub struct DomainSeparator(pub [u8; 32]);
 
 /// Originated from the blockchain transaction input data.
 pub type Calldata = crate::util::Bytes<Vec<u8>>;
+
+/// An on-chain transaction.
+#[derive(Debug)]
+pub struct Transaction {
+    /// The hash of the transaction.
+    pub hash: TxId,
+    /// The address of the solver that submitted the transaction.
+    pub solver: Address,
```

Comment:
If this is "general purpose" type for any regular transaction this should probably be called `from`, right?

Reply:
good one, fixes